regolith/helpers/u_milestonehelper.py

```diff
@@ -102,6 +102,14 @@ def db_updater(self):
                           f"     status: {j.get('status')}")
                 del j['identifier']
             return
+        if rc.type:
+            for k, v in ALLOWED_TYPES.items():
```

Comment:
I am afraid this implementation is a bit confusing.  It may be better to keep all the logic about this together to make the code more readable.  Maybe just validate entries here and exit if they don't validate
```
if rc.status not in [k, v for k, v in ALLOWED_STATI.items()]:
    raise ValueError("blah blah")
```
(something like that, I didn't test it)
then handle how to insert the right thing into the update down below.